src/Build/BackEnd/Components/Communications/NodeManager.cs

```diff
@@ -113,6 +119,11 @@ public NodeInfo CreateNode(NodeConfiguration configuration, NodeAffinity nodeAff
                 nodeId = AttemptCreateNode(_outOfProcNodeProvider, configuration);
             }
 
+            if (nodeId == InvalidNodeId && (nodeAffinity == NodeAffinity.RarNode))
```

Comment:
I want to create only new node in case when is requested RarNode affinity. I don't see any reason to check for NodeAffinity.Any (since it is not flagged enum, there can only be one value).